ra2ce/network/network_wrappers/osm_network_wrapper/osm_network_wrapper.py

```diff
@@ -98,6 +98,29 @@ def get_network_from_polygon(
         _network_wrapper = OsmNetworkWrapper.with_polygon(config_data, polygon)
         return _network_wrapper.get_network()
 
+    @staticmethod
+    def get_network_from_geojson(config_data: NetworkConfigData):
+        """
+        Gets a valid network (`MultiGraph` and `GeoDataFrame`) for the given network configuration,
+        given that its network section contains a valid path for the polygon property.
+
+        Args:
+            config_data (NetworkConfigData): Network data configuration required for OSM download.
+
+        Raises:
+            ValueError: When the `NetworkConfigData` does not contain a valid polygon path
+            for its network section.
+
+        Returns:
+            tuple[MultiGraph, GeoDataFrame]: Resulting network representations.
+        """
+        if not isinstance(config_data.network.polygon, Path):
```

Comment:
Should you also check if the file exists?

Reply:
I could, but seems redundant given the call `_get_clean_graph_from_osm` will already verify this. In fact, after checking for your question I even doubt whether the above's if is even needed.